### PR TITLE
micro optimizations

### PR DESCRIPTION
**Pull request by jsteemann**

### CODE REVIEW

**Comment by jsteemann** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/3017/

**Comment by jsteemann** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/3021/

File: 3rdParty/velocypack/include/velocypack/Slice.h [hunk 1]
```diff
@@ -950,6 +950,71 @@ class Slice {
...
+      if (s == 0) {
+        throw Exception(Exception::InternalError, "Invalid data for Array");
+      }
+      ValueLength end = readIntegerNonEmpty<ValueLength>(_start + 1, offsetSize);
+      return (end - firstSubOffset) / s;
+    } else if (offsetSize < 8) {
+      return readIntegerNonEmpty<ValueLength>(_start + offsetSize + 1, offsetSize);
+    }
+
+    ValueLength end = readIntegerNonEmpty<ValueLength>(_start + 1, offsetSize);
```
Comment (graetzer) [inline 3rdParty/velocypack/include/velocypack/Slice.h]: same calculation as in the first 'if'

File: arangod/RocksDBEngine/RocksDBComparator.cpp [hunk 2]
```diff
@@ -33,17 +33,47 @@
...
+
   int r = memcmp(lhs.data(), rhs.data(), objectIDLength);
+
   if (r != 0) {
+    // different object ID
     return r;
-  } else if (lhs.size() == objectIDLength || rhs.size() == objectIDLength) {
+  } 
+
+  if (ADB_UNLIKELY(lhs.size() == objectIDLength || rhs.size() == objectIDLength)) {
```
Comment (graetzer) [inline arangod/RocksDBEngine/RocksDBComparator.cpp]: Are you sure this is an unlikely branch ?

**Pull request merged**